apps/web/tests/asset-inventory/collector/collector-create/create-collector-case1.spec.ts

```diff
@@ -0,0 +1,331 @@
+            const collectorCount = await page.locator('div.plugin-card-list > div.plugin-card-item').count();
+            const randomIndex = getRandomIndexInRange(collectorCount);
+            await page.getByRole('button', { name: 'Select' }).nth(randomIndex).click();
+            await expect(page.locator('p.step')).toContainText('Step 2');
+        });
+
+        await test.step('7. When [Select Other Plugin] is chosen, confirm Move to Step 1/4.', async () => {
+            await page.getByRole('button', { name: 'Select Other Plugin' }).click();
+            await page.getByRole('button', { name: 'Confirm' }).click();
+            await expect(page.locator('p.step')).toContainText('Step 1');
+        });
+
+        await test.step('8. Click the [Select] Button for the specific plugin again > Confirm Move to Step 2/4.', async () => {
+            await checkApiCall(page, '/repository/plugin/list');
+            const collectorCount = await page.locator('div.plugin-card-list div.plugin-card-item').count();
+            const randomIndex = getRandomIndexInRange(collectorCount);
+            await page.getByRole('button', { name: 'Select' }).nth(randomIndex).click();
+            // await page.getByPlaceholder('Search', { exact: true }).click();
+            // await page.getByPlaceholder('Search', { exact: true }).fill('prowler');
+            // await page.getByPlaceholder('Search', { exact: true }).press('Enter');
+            // await page.getByRole('button', { name: 'Select' }).first().click();
+            await expect(page.locator('p.step')).toContainText('Step 2');
+        });
+
+        let collectorTitle;
+        let selectedPluginName = '';
+        await test.step('9. Fill in the Name field > Select one Provider if applicable.', async () => {
+            let isSameName = false;
+            let checkSameNameCount = 0;
+            const checkSameName = async () => {
+                checkSameNameCount++;
+                const validationText = await page.locator('div.name-form > div.invalid-feedback').textContent();
+                return (validationText ?? '').includes('Name already registered.');
+            };
+            collectorTitle = `test collector-${checkSameNameCount}`;
+            await page.locator('div.name-form > div.p-text-input').getByRole('textbox').fill(collectorTitle);
+            isSameName = await checkSameName();
+            while (isSameName) {
+                collectorTitle = `test collector-${Math.floor(Math.random() * 100)}`;
+                // eslint-disable-next-line no-await-in-loop
+                await page.locator('div.name-form > div.p-text-input').getByRole('textbox').fill(collectorTitle);
+                // eslint-disable-next-line no-await-in-loop
+                isSameName = await checkSameName();
+                if (checkSameNameCount > 10) {
+                    throw new Error('Same name error occurred more than 10 times.');
+                }
+            }
```

Comment:
question: can you explain what is this logic for? Actually.. it's pretty hard to follow up.😅